crates/graphql-mocks/src/dynamic/entity_resolvers.rs

```diff
@@ -0,0 +1,62 @@
+use async_graphql::{dynamic::ResolverContext, ServerError};
+
+pub struct EntityResolverContext<'a> {
+    pub inner_context: &'a ResolverContext<'a>,
+    pub typename: String,
+    pub repr: serde_json::Value,
+}
+
+impl<'a> EntityResolverContext<'a> {
+    pub(super) fn new(inner_context: &'a ResolverContext<'a>, repr: serde_json::Value) -> Self {
+        let typename = repr["__typename"].as_str().expect("a repr must have __typename").into();
+
+        EntityResolverContext {
+            inner_context,
+            typename,
+            repr,
+        }
+    }
+
+    pub fn add_error(&self, error: ServerError) {
+        self.inner_context.query_env.errors.lock().unwrap().push(error);
+    }
+
+    /// Returns the key of the entity we need to resolve
+    pub fn key(&self) -> serde_json::Value {
```

Comment:
Would name it `representation`, it also includes fields from `@requires` not just from `@key`

Reply:
hmm, fair point.  Although the representation usually also includes the __typename....

Reply:
Tempted to just not offer this as a convenience

Reply:
ah didn't realize it, could just remove it yeah

Reply:
yeah, just going to bin this.  Could implement something that looks at the actual keys defined for the entity, but not worth it unless we need it.